### PR TITLE
Improve k8s dashboard

### PR DESCRIPTION
**Pull request by ChrsMark**
## What does this PR do?
This PR leverages https://github.com/elastic/kibana/pull/126015 in order to solve https://github.com/elastic/integrations/issues/2159

## Why is it important?
At the moment the k8s Dashboard can be showing inconsistent values in the views for the "Desired", "Available" and "Unavailable" Pods. The reason is described at https://github.com/elastic/integrations/issues/2159. With this chance we aim to improve the situation so as the views to be showing correct values by using the proper aggregations.

## Related issues

- Relates https://github.com/elastic/integrations/issues/2159

## Screenshots

Changing the time range of the dashboard was giving inconsistent values in the views. Below find attached the consistent views with different time ranges:
Screenshot 2022-03-18 at 12 13 17 PM
Screenshot 2022-03-18 at 12 13 13 PM
Screenshot 2022-03-18 at 12 13 07 PM

Screenshot 2022-03-18 at 12 09 41 PM

### CODE REVIEW

File: metricbeat/module/kubernetes/_meta/kibana/7/visualization/da1ff7c0-30ed-11e7-b9e5-2b5b07213ab3-ecs.json [hunk 1]
```diff
@@ -40,47 +47,68 @@
                 "id": "2fe9d3b0-30d5-11e7-8df8-6d3604a72912",
                 "index_pattern": "metricbeat-*",
                 "interval": "auto",
+                "isModelInvalid": false,
```
Comment (tetianakravchenko) [inline metricbeat/module/kubernetes/_meta/kibana/7/visualization/da1ff7c0-30ed-11e7-b9e5-2b5b07213ab3-ecs.json:50]: is this field part of `Panel Option`? could you add a screenshot for the page where this is defined?

**Comment by ChrsMark** [comment]
@tetianakravchenko it seems that `drop_last_bucket` value is locked when choosing `Entire time range`:


Screenshot 2022-03-22 at 12 13 31 PM

File: metricbeat/module/kubernetes/_meta/kibana/7/visualization/da1ff7c0-30ed-11e7-b9e5-2b5b07213ab3-ecs.json (hunk 1 above)
Comment (ChrsMark) [reply metricbeat/module/kubernetes/_meta/kibana/7/visualization/da1ff7c0-30ed-11e7-b9e5-2b5b07213ab3-ecs.json:50]: I cannot find it in the view tbh, @flash1293 do you have an answer handy for this?

**Review by tetianakravchenko — APPROVED** [review]
LGTM, just for curiosity would like to know where `"isModelInvalid": false` in UI is defined

**Comment by flash1293** [comment]
@ChrsMark @tetianakravchenko @kvch I thought a bit more about this and I'm not 100% sure anymore whether users would expect the "entire time range" mode for this one. AFAIK in most beats dashboards the metric numbers are meant to show the "current state" of the system, but by using entire time range with the series agg and multi field group by, it will take into account "stale" information as well.

This is the case I'm a bit worried about:
* User has multiple deployments in multiple namespaces running, for each of them a new document is indexed every n seconds
* They look at the dashboard, timerange configured to 15 minutes
* Metric on the dashboard is consistent, no matter the time range (because it's always ever taking into account the last value per time series)
* User deletes an unhealthy deployment which caused a lot of "desired pods"
* User refreshes the dashboard after a minute or so
* They expect the "desired pods" metric to drop (because the deployment was deleted after all)
* The number stays the same, it only drops after 15 minutes because even though no new documents for the deleted deployment were indexed, there was still a "last value" within the last 15 minutes which was rolled up into the metric

Another weird case:
* They look at the dashboard, timerange configured to 15 minutes
* They delete a deployment, come back to the dashboard after 30 minutes
* Everything shows consistent, because the deleted deployment doesn't have a "last value" in the last 15 minutes 
* They increase the time range to two hours to check something
* "Desired pods" metric goes up
* User is confused - why are there more "desired pods" if I'm looking at more data in the past???


These cases could be improved by using "last value mode" with "drop last bucket" and an interval which is larger than the polling interval (let's say 2 minutes). By doing this, only the deployments which got documents indexed in the last 2 minutes will be rolled up into the final metric instead of all of them for the entire time range, resulting in a much faster reflection of K8s state changes at the expense of a more expensive query and some hidden magic (the 2 minutes are not visible to the user in any way while the entire time range is known).

I'm not sure about this, in the end it comes down to what the user expects this number is representing - the state of the system by the end of the configured time range or the state of the system "during" the whole time range. Your decision, just wanted to make this transparent.

**Comment by ChrsMark** [comment]
Thanks for your feedback @flash1293, I think that the goal of these visualisations is to show the "current" state of the cluster no matter what is the time range. So I would say that using `"last value mode" with "drop last bucket"` is better option for this. I remember that during our chat you mentioned about this option having some drawbacks but I do not remember exactly what was it. If we agree that `"last value mode" with "drop last bucket"` is better for showing the current cluster status regardless of the time range then I can move on and open fixup PR.

**Comment by ChrsMark** [comment]
I made some checks again and it seems to work correctly with the `"last value mode" with "drop last bucket"`. So I'm +1 on changing this and document the detail about the interval so as users be able to update it in case they have bigger collection periods.

**Comment by ChrsMark** [comment]
@flash1293 if the collection period is 5 minutes a user also need to increase the `interval` too or that's not a problem?

**Comment by flash1293** [comment]
@ChrsMark If the interval is smaller than the collection period it will be possible they wont see any data if they are unlucky.

**Comment by ChrsMark** [comment]
> @ChrsMark If the interval is smaller than the collection period it will be possible they wont see any data if they are unlucky.

Cool, I will have it documented then -> https://github.com/elastic/beats/pull/30986

**Pull request merged**